core/src/main/java/org/apache/iceberg/GenericManifestFile.java

```diff
@@ -46,12 +46,14 @@
   private String manifestPath = null;
   private Long length = null;
   private int specId = -1;
+  private long sequenceNumber = 0;
+  private long minSequenceNumber = 0;
   private Long snapshotId = null;
   private Integer addedFilesCount = null;
-  private Long addedRowsCount = null;
   private Integer existingFilesCount = null;
-  private Long existingRowsCount = null;
   private Integer deletedFilesCount = null;
+  private Long addedRowsCount = null;
```

Comment:
Yes. I think it's likely that these are going to be null in some cases, like manifests that contain equality delete files. Instead of mixing null, non-null, null, non-null, etc. I think it's better to keep the probably-null columns colocated for compression.